### DISCUSSION TITLE
Quarks startup and shutdown events are not working key-cloak 23 onwards

### DISCUSSION THREAD

**Discussion by tharunjasti** [keycloak/keycloak#30061] [Archive]
### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

core

### Describe the bug

Hi,

We previously used both a Quarkus StartupEvent, ShutdownEvent, this used to work in 20V, but these are not working in 23V,
I've been trying to update from 20.0.3 to 23.0.5, but I'm running into the issue that these no longer work.

https://quarkus.io/guides/lifecycle#startup_annotation

I am assuming this is a bug, i.e. that keycloak would like to support [Listening for startup and shutdown events](https://quarkus.io/guides/lifecycle#startup_annotation). If not, my apologies.

Currently we are doing work around using RealmResourceProviderFactory postInit method.

### Version

23.0.5

### Regression

- [X] The issue is a regression

### Expected behavior

StartupEvent and ShutdownEvent events should trigger

### Actual behavior

StartupEvent and ShutdownEvent events are not triggering

### How to Reproduce?

Build a simple Logging class like this  https://quarkus.io/guides/lifecycle#startup_annotation
Package it into a jar file.
Copy jar file to Keycloak Providers directory
Start Keycloak and check below logs.
Be notified when the application starts
Be notified when the application stops

@ApplicationScoped
public class AppLifecycleBean {

    private static final Logger LOGGER = Logger.getLogger("ListenerBean");

    void onStart(@Observes StartupEvent ev) {               
        LOGGER.info("The application is starting...");
    }

    void onStop(@Observes ShutdownEvent ev) {               
        LOGGER.info("The application is stopping...");
    }

}

**Comment by shawkins** [comment]
> that keycloak would like to support [Listening for startup and shutdown events](https://quarkus.io/guides/lifecycle#startup_annotation). If not, my apologies.

Keycloak is hesitant to expose quarkus features as supported by keycloak. That is why the quarkus.properties file is still considered unsupported. 

So this is a grey area, but I don't see any reason why these events shouldn't work.

> Copy jar file to Keycloak Providers directory

Are you including a jandex index or a beans.xml in this jar?

**Reply by tharunjasti** [reply]
i had added code in this repo.
https://github.com/tharunjasti/keycloak_code_snippets/tree/main
same code startuo events is not triggering for keycloak 23
